azext_iot/tests/test_iot_ext_unit.py

```diff
@@ -469,6 +496,142 @@ def test_device_update_error(self, serviceclient_generic_error, req):
+                        "type": "sas",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                ), True, "enabled", None, "x509_ca", "None", "None", None, None
+            ),
+            generate_device_custom(
+                generate_device_show(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "123", "secondaryKey": "321"},
+                        "type": "sas",
+                    }
+                ), True, "enabled", None, None, None, None, "primary_Updated", "secondary_Updated"
+            ),
+            generate_device_custom(
+                generate_device_show(
+                    authentication={
+                        "type": "x509_thumbprint",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                ), True, "enabled", None, None, "primaryThumbprint_Updated", "secondaryThumbprint_Updated", None, None
+            ),
+        ]
+    )
+    def test_iot_device_custom(fixture_cmd, req):
+        instance = subject.update_iot_device_custom(
+            req["device"],
+            req["ee"],
+            req["status"],
+            req["status_reason"],
+            req["auth"],
+            req["ptp"],
+            req["stp"],
+            req["pk"],
+            req["sk"],
+        )
+        assert instance["capabilities"]["iotEdge"] == req["ee"]
+        assert instance["status"] == req["status"]
+        assert instance["statusReason"] == req["status_reason"]
+
+        if req["auth"] == "shared_private_key":
+            assert instance['authentication']['symmetricKey']['primaryKey'] == req["pk"]
+            assert instance['authentication']['symmetricKey']['secondaryKey'] == req["sk"]
+            assert instance['authentication']['type'] == 'sas'
+
+        if req["auth"] == "x509_thumbprint":
+            assert instance['authentication']['x509Thumbprint']['primaryThumbprint'] == req["ptp"]
+            assert instance['authentication']['x509Thumbprint']['secondaryThumbprint'] == req["stp"]
+            assert instance['authentication']['type'] == 'selfSigned'
+
+        if req["auth"] == "x509_ca":
+            assert instance['authentication']['type'] == "certificateAuthority"
+
+        if req["auth"] is None and (req["pk"] or req["sk"] or req["ptp"] or req["stp"]):
+            if instance['authentication']['type'] == 'sas':
+                assert instance['authentication']['symmetricKey']['primaryKey'] == req["pk"]
+                assert instance['authentication']['symmetricKey']['secondaryKey'] == req["sk"]
+            if instance['authentication']['type'] == 'selfSigned':
+                assert instance['authentication']['x509Thumbprint']['primaryThumbprint'] == req["ptp"]
+                assert instance['authentication']['x509Thumbprint']['secondaryThumbprint'] == req["stp"]
```

Comment:
What will happen we won't pass secondary thumbprint or req["stp"]=None, then it will fail